src/elm/arithmetic.js

```diff
@@ -23,6 +24,8 @@ class Add extends Expression {
     const sum = args.reduce((x, y) => {
       if (x.isQuantity || x.isDateTime || x.isDate || (x.isTime && x.isTime())) {
         return doAddition(x, y);
+      } else if (x.isUncertainty && y.isUncertainty) {
+        return new Uncertainty(x.low + y.low, x.high + y.high);
```

Comment:
I think we also need to test what _type_ of uncertainty this is.  If it is an uncertainty with dates as the low/high, for example, this would not work.  So you probably need to check the point types too.  To cover the bases, I guess it might look like:
```
} else if (x.isUncertainty && y.isUncertainty) {
  if (x.low.isQuantity || x.low.isDateTime || x.low.isDate || (x.low.isTime && x.low.isTime()) {
    return new Uncertainty(doAddition(x.low, y.low), doAddition(x.high, y.high));
  } else {
    return new Uncertainty(x.low + y.low, x.high + y.high);
  }
} else {
```

Reply:
And this same approach would likely be appropriate for the others.